src/main/java/core/basesyntax/service/impl/StrategyOperationServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.OperationType;
+import core.basesyntax.service.StrategyOperationService;
+import core.basesyntax.strategy.OperationHandler;
+import exception.OperationException;
+import java.util.Map;
+
+public class StrategyOperationServiceImpl implements StrategyOperationService {
+    private Map<OperationType, OperationHandler> operationHandlerMap;
+
+    public StrategyOperationServiceImpl(Map<OperationType, OperationHandler> operationHandlerMap) {
+        this.operationHandlerMap = operationHandlerMap;
+    }
+
+    @Override
+    public OperationHandler get(OperationType operationType) {
+        if (operationHandlerMap.get(operationType) != null) {
```

Comment:
You can simplify the check by using `operationHandlerMap.containsKey(operationType)` to verify if the map contains the key before retrieving the handler.